0000-product.md

```diff
@@ -0,0 +1,463 @@
+about the product, such as the product’s name and description.  Therefore,
+Product becomes a reusable and important component within Grid.
+
+# Guide-Level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Entities
+
+A **__product__** is an archetype of an item that is transacted, traded, or
+referenced in a supply chain.  Each product has a **__product_namespace__**.
+This RFC defines a single product_namespace for GS1; a product in the GS1
+product_namespace is called a GS1 product.  Note that this design supports
+extending additional product namespaces in the future.
+
+A product is referenced using a **product_id**.  For GS1 products, the
+product_id is a Global Trade Item Number (GTIN), which is part of the GS1
+specifications.
+
+Each product has an **__owner__**, which is the identifier of the organization
+responsible for maintaining the product.  An **__agent__** acts on behalf of an
+organization, and all permissions are defined in terms of the actions agents can
+perform.  Organizations and agents are defined and managed by Pike (another
+component of Grid).
+
+A product has one or more **properties**.  Properties are described in the Grid
+Primitives RFC.  A *property namespace* contains multiple *property schemas*.  A
+property schema associates a name (such as “length”) with a data type (such as
+integer).  GS1 products may only include properties defined in the GS1 product
+property namespace.
+
+## Transactions
+
+Products are managed by submitting transactions to Hyperledger Grid, which will
+process them with the Grid Product smart contract. The following transactions
+are supported:
+
+* ProductCreate - create a Product and store it in state.  
+* ProductUpdate - update (replace) the properties of a Product already in
+  state.
+* ProductDelete - remove a Product from state.
+
+## Permissions
+
+Creation of GS1 products is restricted to agents acting on behalf of the
+organization that is associated with the GS1 company prefix encoded in the GTIN.
+An organization must have the GS1 company prefix in the “gs1_company_prefixes”
+metadata field for its Pike organization.  (Organization-level metadata will
+need to be implemented in Pike.) When a product is created, its owning
+organization is stored with the product in an “owner” field.
+
+Updates to products is restricted to agents acting on behalf of the organization
+stored in the product’s owner field.  Only property fields can be updated.
+Product_id, product_namespace, and owner fields are immutable.
+
+Deletion of products is restricted to agents acting on behalf of the
+organization in the product’s owner field.  A setting will turn off deletion
+entirely, with the intent that it could be enabled/disabled by a Grid
+administrator.  Disabling delete is useful because external systems may have
+references to these products and deleting them could leave dangling references.
+
```

Comment:
Products very often get delisted. In Healthcare, it is extremely valuable and required to understand when a product has been consumed or should not be expected to be seen in the supply chain again. This exact scenario was accommodated for in EPCIS. If we are just talking about master data of a class level GTIN - OK.. but if we are talking about lower levels (batch/lot or serial) then I question this. Understanding when something has been deleted or removed from a supply chain may have extreme value - especially if later down the line you see it reintroduced. Perhaps having the original creation will throw the same flag if it is created again?? But what about updates?

Reply:
We are talking about master data of a class level GTIN.

This implementation of Product is simply a way to represent and an item within Grid. I think that a grid application, like track and trace, can leverage Product to provide that level of tracking. This product RFC is not intended to provide application-level functionality and is describing Product as more of a base model/primitive in the Grid system, to enable the construction of more complex applications.

Reply:
Agree - @genasmorgan concerns seem to be scoped to SGTIN and/or other lot/batch level stuff.

Reply:
add a disclaimer about delete function